java/integTest/src/test/java/glide/standalone/StandaloneClientTests.java

```diff
@@ -159,4 +166,120 @@ public void closed_client_throws_ExecutionException_with_ClosingException_as_cau
+            var emptyPasswordException =
+                    assertThrows(
+                            ExecutionException.class, () -> testClient.updateConnectionPassword("", true).get());
+            assertInstanceOf(RequestException.class, emptyPasswordException.getCause());
+
+            var noPasswordException =
+                    assertThrows(
+                            ExecutionException.class, () -> testClient.updateConnectionPassword(true).get());
+            assertInstanceOf(RequestException.class, noPasswordException.getCause());
+        }
+    }
+
+    @SneakyThrows
+    @Test
+    public void update_connection_password_no_server_auth() {
+        var pwd = UUID.randomUUID().toString();
+
+        try (var testClient = GlideClient.createClient(commonClientConfig().build()).get()) {
+            // validate that we can use the client
+            assertNotNull(testClient.info().get());
+
+            // Test that immediate re-authentication fails when no server password is set.
+            var exception =
+                    assertThrows(
+                            ExecutionException.class, () -> testClient.updateConnectionPassword(pwd, true).get());
+            assertInstanceOf(RequestException.class, exception.getCause());
+        }
+    }
+
+    @SneakyThrows
+    @Test
+    public void update_connection_password_long() {
+        var pwd = RandomStringUtils.randomAlphabetic(1000);
+
+        try (var testClient = GlideClient.createClient(commonClientConfig().build()).get()) {
+            // validate that we can use the client
+            assertNotNull(testClient.info().get());
+
+            // Test replacing connection password with a long password string.
+            assertEquals(OK, testClient.updateConnectionPassword(pwd, false).get());
+        }
+    }
+
+    @Timeout(50)
+    @SneakyThrows
+    @Test
+    public void replace_password_immediateAuth_wrong_password() {
+        var pwd = UUID.randomUUID().toString();
+        var notThePwd = UUID.randomUUID().toString();
+
+        GlideClient adminClient = GlideClient.createClient(commonClientConfig().build()).get();
+        try (var testClient = GlideClient.createClient(commonClientConfig().build()).get()) {
+            // validate that we can use the client
+            assertNotNull(testClient.info().get());
+
+            // set the password to something else
+            adminClient.configSet(Map.of("requirepass", notThePwd)).get();
+
+            // Test that re-authentication fails when using wrong password.
+            var exception =
+                    assertThrows(
+                            ExecutionException.class, () -> testClient.updateConnectionPassword(pwd, true).get());
+            assertInstanceOf(RequestException.class, exception.getCause());
+
+            // But using something else password returns OK
+            assertEquals(OK, testClient.updateConnectionPassword(notThePwd, true).get());
+        } finally {
```

Comment:
Why don't we have finally block in the first 3 tests?

Reply:
the adminClient needs to be cleaned up in this test (only).

Reply:
The `try` cleans up the testClient